File: backend/pkg/gameplugin/sdk/game-plugin/src/transport.ts
import { Envelope } from './protocol';

export interface Transport {
  send(message: Envelope): Promise<void>;
  receive(): Promise<Envelope>;
  close(): Promise<void>;
  // Optional fast path used by stdio transports. Other transports can omit it
  // and the SDK falls back to bounded JSON/base64 binary.write requests.
  sendBinaryFrame?(message: Envelope, objectId: string, offset: number, data: Uint8Array): Promise<void>;
}

File: sdk/game-plugin-go/transport.go
package sdk

import (
	"context"

	"github.com/u-ai/game-plugin-sdk-go/protocol"
)

type Transport interface {
	Send(ctx context.Context, message protocol.Envelope) error
	Receive(ctx context.Context) (protocol.Envelope, error)
	Close() error
}

type BinaryFrameTransport interface {
	Transport
	SendBinaryFrame(ctx context.Context, message protocol.Envelope, objectID string, offset int64, data []byte) error
}
